nisystemlink/clients/testmonitor/models/_result.py

```diff
@@ -0,0 +1,90 @@
+from datetime import datetime
+from enum import Enum
+from typing import Dict, List, Optional
+
+from nisystemlink.clients.core._uplink._json_model import JsonModel
+from pydantic import Extra
+
+
+class StatusType(str, Enum):
+    """The types of statuses that a result can have."""
+
+    LOOPING = "LOOPING"
+    SKIPPED = "SKIPPED"
+    CUSTOM = "CUSTOM"
+    DONE = "DONE"
+    PASSED = "PASSED"
+    FAILED = "FAILED"
+    RUNNING = "RUNNING"
+    WAITING = "WAITING"
+    TERMINATED = "TERMINATED"
+    ERRORED = "ERRORED"
+    TIMED_OUT = "TIMED_OUT"
+
+
+class StatusObject(JsonModel):
+    """Contains information about a status object."""
+
+    status_type: StatusType
+    """The type of status."""
+
+    status_name: Optional[str]
+    """The name of the status."""
+
```

Comment:
Create a constant class for all standard `StatusType` as in most use cases users don't use the `status_name` and it is an additional step to create a `StatusObject`

```suggestion
class StandardResultStatus:
    LOOPING = StatusObject(StatusType.LOOPING)
    SKIPPED = StatusObject(StatusType.SKIPPED)
    ...
   
```

Reply:
Added as suggested

Reply:
Moved to `Status` class